Resync web-platform-tests/resources

aprotyas opened the pull request:
#### fdcc963be479c8f0fa2c36e87051068f9c3bacff
<pre>
Resync web-platform-tests/resources
<a href="https://bugs.webkit.org/show_bug.cgi?id=294104">https://bugs.webkit.org/show_bug.cgi?id=294104</a>
<a href="https://rdar.apple.com/152695590">rdar://152695590</a>

Reviewed by NOBODY (OOPS!).

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/8e564b77589e78ad1f3189a88b4ea73c66f34bf0">https://github.com/web-platform-tests/wpt/commit/8e564b77589e78ad1f3189a88b4ea73c66f34bf0</a>

We have not imported the `web-platform-tests/resources` folder in a
while. This patch addresses that need.

* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout-th.js:
(window.checkLayout):
* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/declarative-shadow-dom-polyfill.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness-shadowrealm.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness.js:
(IdlArray.prototype.is_json_type):
(IdlArray.prototype.test):
(IdlArray.prototype.merge_mixins):
(IdlInterface.prototype.test):
(IdlInterface.prototype.do_member_operation_asserts):
(IdlNamespace.prototype.test):
(idl_test):
* LayoutTests/imported/w3c/web-platform-tests/resources/test/conftest.py:
(pytest_collect_file):
(pytest_configure):
(HTMLFile):
(HTMLFile.__init__):
(HTMLFile.collect):
(HTMLItem):
(HTMLItem.__init__):
(HTMLItem.reportinfo):
(HTMLItem.runtest):
(HTMLItem._assert_sequence):
(HTMLItem.repr_failure): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/abortsignal.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/api-tests-1.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/assert-throws-dom.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/idlharness/IdlDictionary/test_partial_interface_of.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/idlharness/IdlInterface/test_partial_interface_of.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/idlharness/IdlNamespace/test_operation.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/step_wait.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/unit/IdlArray/is_json_type.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/unit/late-test.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tox.ini:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver.js:
(const.features):
(assertBidiIsEnabled):
(window.test_driver.bidi.bluetooth.handle_request_device_prompt):
(window.test_driver.bidi.bluetooth.simulate_adapter):
(window.test_driver.bidi.bluetooth.disable_simulation):
(window.test_driver.bidi.bluetooth.simulate_preconnected_peripheral):
(window.test_driver.bidi.bluetooth.simulate_gatt_connection_response):
(window.test_driver.bidi.bluetooth.simulate_gatt_disconnection):
(window.test_driver.bidi.bluetooth.request_device_prompt_updated.subscribe):
(window.test_driver.bidi.bluetooth.request_device_prompt_updated):
(window.test_driver.bidi.bluetooth.):
(window.test_driver.bidi.):
(window.test_driver):
(window.test_driver.set_test_context): Deleted.
(window.test_driver.message_test): Deleted.
(window.test_driver.bless): Deleted.
(window.test_driver.click): Deleted.
(window.test_driver.delete_all_cookies): Deleted.
(window.test_driver.get_all_cookies): Deleted.
(window.test_driver.async let): Deleted.
(window.test_driver.get_named_cookie): Deleted.
(window.test_driver.send_keys): Deleted.
(window.test_driver.freeze): Deleted.
(window.test_driver.minimize_window): Deleted.
(window.test_driver.set_window_rect): Deleted.
(window.test_driver.get_window_rect): Deleted.
(window.test_driver.action_sequence): Deleted.
(window.test_driver.generate_test_report): Deleted.
(window.test_driver.set_permission): Deleted.
(window.test_driver.add_virtual_authenticator): Deleted.
(window.test_driver.remove_virtual_authenticator): Deleted.
(window.test_driver.add_credential): Deleted.
(window.test_driver.get_credentials): Deleted.
(window.test_driver.remove_credential): Deleted.
(window.test_driver.remove_all_credentials): Deleted.
(window.test_driver.set_user_verified): Deleted.
(window.test_driver.set_storage_access): Deleted.
(window.test_driver.set_spc_transaction_mode): Deleted.
(window.test_driver.set_rph_registration_mode): Deleted.
(window.test_driver.cancel_fedcm_dialog): Deleted.
(window.test_driver.click_fedcm_dialog_button): Deleted.
(window.test_driver.select_fedcm_account): Deleted.
(window.test_driver.get_fedcm_account_list): Deleted.
(window.test_driver.get_fedcm_dialog_title): Deleted.
(window.test_driver.get_fedcm_dialog_type): Deleted.
(window.test_driver.set_fedcm_delay_enabled): Deleted.
(window.test_driver.reset_fedcm_cooldown): Deleted.
(window.test_driver.create_virtual_sensor): Deleted.
(window.test_driver.update_virtual_sensor): Deleted.
(window.test_driver.remove_virtual_sensor): Deleted.
(window.test_driver.get_virtual_sensor_information): Deleted.
(window.test_driver.set_device_posture): Deleted.
(window.test_driver.clear_device_posture): Deleted.
(window.test_driver_internal.async click): Deleted.
(window.test_driver_internal.async delete_all_cookies): Deleted.
(window.test_driver_internal.async get_all_cookies): Deleted.
(window.test_driver_internal.async get_named_cookie): Deleted.
(window.test_driver_internal.): Deleted.
(window.test_driver_internal.async send_keys): Deleted.
(window.test_driver_internal.async freeze): Deleted.
(window.test_driver_internal.async minimize_window): Deleted.
(window.test_driver_internal.async set_window_rect): Deleted.
(window.test_driver_internal.async get_window_rect): Deleted.
(window.test_driver_internal.async action_sequence): Deleted.
(window.test_driver_internal.async generate_test_report): Deleted.
(window.test_driver_internal.async set_permission): Deleted.
(window.test_driver_internal.async add_virtual_authenticator): Deleted.
(window.test_driver_internal.async remove_virtual_authenticator): Deleted.
(window.test_driver_internal.async add_credential): Deleted.
(window.test_driver_internal.async get_credentials): Deleted.
(window.test_driver_internal.async remove_credential): Deleted.
(window.test_driver_internal.async remove_all_credentials): Deleted.
(window.test_driver_internal.async set_user_verified): Deleted.
(window.test_driver_internal.async set_storage_access): Deleted.
(window.test_driver_internal.async set_spc_transaction_mode): Deleted.
(window.test_driver_internal.set_rph_registration_mode): Deleted.
(window.test_driver_internal.async cancel_fedcm_dialog): Deleted.
(window.test_driver_internal.async click_fedcm_dialog_button): Deleted.
(window.test_driver_internal.async select_fedcm_account): Deleted.
(window.test_driver_internal.async get_fedcm_account_list): Deleted.
(window.test_driver_internal.async get_fedcm_dialog_title): Deleted.
(window.test_driver_internal.async get_fedcm_dialog_type): Deleted.
(window.test_driver_internal.async set_fedcm_delay_enabled): Deleted.
(window.test_driver_internal.async reset_fedcm_cooldown): Deleted.
(window.test_driver_internal.async create_virtual_sensor): Deleted.
(window.test_driver_internal.async update_virtual_sensor): Deleted.
(window.test_driver_internal.async remove_virtual_sensor): Deleted.
(window.test_driver_internal.async get_virtual_sensor_information): Deleted.
(window.test_driver_internal.async set_device_posture): Deleted.
(window.test_driver_internal.async clear_device_posture): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness-shadowrealm-audioworkletprocessor.js: Added.
(globalThis.TestRunner):
(globalThis.TestRunner.prototype.fetchOverPortExecutor):
(globalThis.TestRunner.prototype.async createShadowRealmAndStartTests):
(globalThis.TestRunner.prototype.process):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness-shadowrealm-inner.js: Added.
(globalThis.setShadowRealmGlobalProperties):
(globalThis.GLOBAL.isWindow):
(globalThis.GLOBAL.isWorker):
(globalThis.GLOBAL.isShadowRealm):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness-shadowrealm-outer.js: Added.
(globalThis.shadowRealmEvalAsync):
(async globalThis):
(async if):
(globalThis.getPostMessageFunc):
(else):
(realm.evaluate):
(globalThis.setupFakeDynamicImportInShadowRealm):
(globalThis.setupFakeFetchOverMessagePort):
(globalThis.createSetupErrorResult):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness.js:
(WindowTestEnvironment):
(WindowTestEnvironment.prototype.setup_messages):
(WindowTestEnvironment.prototype.test_timeout):
(ShellTestEnvironment):
(ShellTestEnvironment.prototype.next_default_test_name):
(is_service_worker):
(promise_rejects_js):
(promise_rejects_dom):
(promise_rejects_exactly):
(this.wait_for):
(this.stop_watching):
(EventWatcher):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharnessreport.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/web-bluetooth-bidi-test.js: Added.
(string_appeared_here.convertToBidiManufacturerData):
(FakeBluetooth):
(FakeBluetooth.prototype.async simulateCentral):
(FakeCentral):
(FakeCentral.prototype.async simulatePreconnectedPeripheral):
(FakeCentral.prototype.fetchOrCreatePeripheral_):
(FakePeripheral):
(initializeBluetoothBidiResources):
* LayoutTests/resources/testharness.js:
(WindowTestEnvironment):
(WindowTestEnvironment.prototype._dispatch):
(WindowTestEnvironment.prototype.setup_messages):
(WindowTestEnvironment.prototype.test_timeout):
(ShadowRealmTestEnvironment):
(ShadowRealmTestEnvironment.prototype.begin):
(ShadowRealmTestEnvironment.prototype.add_on_loaded_callback):
(ShellTestEnvironment):
(ShellTestEnvironment.prototype.next_default_test_name):
(create_test_environment):
(is_service_worker):
(promise_rejects_js):
(promise_rejects_dom):
(promise_rejects_exactly):
(this.wait_for):
(this.stop_watching):
(EventWatcher):
(promise_setup):
(done): Deleted.
(generate_tests): Deleted.
(step_timeout): Deleted.
(truncate): Deleted.
(is_node): Deleted.
(format_value): Deleted.

Replace the now outdated copy under LayoutTests/resources.

* LayoutTests/tests-options.json:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fdcc963be479c8f0fa2c36e87051068f9c3bacff

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107250 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26932 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17336 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112464 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57783 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27613 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35432 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81414 "Found 105 new test failures: fast/css/Element-style.html http/wpt/push-api/pushManager.any.html http/wpt/push-api/pushManager.any.serviceworker.html http/wpt/webxr/xrDevice_requestSession_previously_enabled_feature_not_requested.https.html imported/w3c/web-platform-tests/WebIDL/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/compression/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-002.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-003.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-004.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-005.html ... (failure)") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110179 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21898 "Found 4 new test failures: fast/css/Element-style.html fast/events/ios/select-all-with-existing-selection.html http/tests/security/cross-origin-blob-transfer.html webrtc/legacy-api.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96669 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61783 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21324 "Found 60 new test failures: imported/w3c/web-platform-tests/WebIDL/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.sharedworker.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.worker.html imported/w3c/web-platform-tests/beacon/beacon-cors.https.window.html imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-basics.https.html imported/w3c/web-platform-tests/compression/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/credential-management/credentialscontainer-get-basics.https.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-002.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-003.html ... (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14804 "Found 60 new test failures: fast/css/Element-style.html fonts/monospace.html fonts/sans-serif.html fonts/serif.html fonts/unicode-character-font-crash.html fonts/use-typo-metrics-1.html http/tests/security/contentSecurityPolicy/block-all-mixed-content/insecure-script-in-iframe.html http/tests/security/contentSecurityPolicy/upgrade-insecure-requests/proper-nested-upgrades.html http/tests/security/cookie-module-propagate.html http/tests/security/cookie-module.html ... (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57229 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91253 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14833 "Found 60 new test failures: accessibility/mac/element-focus.html compositing/tiling/empty-to-tiled.html compositing/transforms/transformed-replaced-with-shadow-children.html fast/css/Element-style.html fast/forms/select-size.html http/tests/multipart/multipart-async-image.html http/tests/security/cross-origin-blob-transfer.html imported/w3c/web-platform-tests/WebIDL/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.sharedworker.html ... (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115564 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34316 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25287 "Found 60 new test failures: accessibility/mac/element-focus.html fast/css/Element-style.html http/tests/images/gif-progressive-load.html http/tests/security/cross-origin-blob-transfer.html imported/w3c/web-platform-tests/WebIDL/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.sharedworker.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.worker.html imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-basics.https.html imported/w3c/web-platform-tests/compression/idlharness-shadowrealm.window.html ... (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90456 "Found 105 new test failures: fast/css/Element-style.html http/tests/security/cross-origin-blob-transfer.html http/wpt/push-api/pushManager.any.html http/wpt/push-api/pushManager.any.serviceworker.html imported/w3c/web-platform-tests/WebIDL/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-basics.https.html imported/w3c/web-platform-tests/compression/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-002.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-003.html imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-004.html ... (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34692 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92918 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90188 "Passed tests") | 
| | [❌ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35095 "Found 1 new test failure: imported/w3c/web-platform-tests/webxr/xrDevice_requestSession_requiredFeatures_unknown.https.html (failure)") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12901 "Found 60 new test failures: accessibility/mac/element-focus.html compositing/geometry/layer-position-after-removing-perspective.html compositing/transforms/transformed-replaced-with-shadow-children.html fast/css/Element-style.html http/tests/security/cross-origin-blob-transfer.html imported/w3c/web-platform-tests/WebIDL/idlharness-shadowrealm.window.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.sharedworker.html imported/w3c/web-platform-tests/background-fetch/idlharness.https.any.worker.html imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-basics.https.html ... (failure)") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30046 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34238 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33984 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37339 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35645 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/resources/testharnessreport.js:
We need this for our reporting - when we sync, it gets deleted, so we need to undo this one.

aprotyas replied:
My bad, I'll fix it!

aprotyas replied:
Fixed in https://github.com/WebKit/WebKit/commit/1764e8e7af00f339bb5889bd8aabab5cc59b4162

gsnedders replied:
No! This should be sync'd; our `testharnessreport.js` lives in `LayoutTests/resources`, and we override it in the `config.json`. There shouldn't be anything excluded.

Ahmad-S792 replied:
Thanks @gsnedders - I was not aware of it.

gsnedders reviewed (changes requested):
I tried to do this a month ago, and then ran into us having our own `testdriver-vendor.js` there, which I started to fix in https://github.com/WebKit/WebKit/pull/44728 (but then found more places that needed changed!).

Lemme try and fix that, then you can try reimporting resources again.

aprotyas committed:
Resync web-platform-tests/resources
https://bugs.webkit.org/show_bug.cgi?id=294104
rdar://152695590

Reviewed by NOBODY (OOPS!).

Upstream commit: https://github.com/web-platform-tests/wpt/commit/8e564b77589e78ad1f3189a88b4ea73c66f34bf0

We have not imported the `web-platform-tests/resources` folder in a
while. This patch addresses that need.

* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout-th.js:
(window.checkLayout):
* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/declarative-shadow-dom-polyfill.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness-shadowrealm.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness.js:
(IdlArray.prototype.is_json_type):
(IdlArray.prototype.test):
(IdlArray.prototype.merge_mixins):
(IdlInterface.prototype.test):
(IdlInterface.prototype.do_member_operation_asserts):
(IdlNamespace.prototype.test):
(idl_test):
* LayoutTests/imported/w3c/web-platform-tests/resources/test/conftest.py:
(pytest_collect_file):
(pytest_configure):
(HTMLFile):
(HTMLFile.__init__):
(HTMLFile.collect):
(HTMLItem):
(HTMLItem.__init__):
(HTMLItem.reportinfo):
(HTMLItem.runtest):
(HTMLItem._assert_sequence):
(HTMLItem.repr_failure): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/abortsignal.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/api-tests-1.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/assert-throws-dom.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/idlharness/IdlDictionary/test_partial_interface_of.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/idlharness/IdlInterface/test_partial_interface_of.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/idlharness/IdlNamespace/test_operation.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/step_wait.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/functional/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/unit/IdlArray/is_json_type.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tests/unit/late-test.html:
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tox.ini:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver.js:
(const.features):
(assertBidiIsEnabled):
(window.test_driver.bidi.bluetooth.handle_request_device_prompt):
(window.test_driver.bidi.bluetooth.simulate_adapter):
(window.test_driver.bidi.bluetooth.disable_simulation):
(window.test_driver.bidi.bluetooth.simulate_preconnected_peripheral):
(window.test_driver.bidi.bluetooth.simulate_gatt_connection_response):
(window.test_driver.bidi.bluetooth.simulate_gatt_disconnection):
(window.test_driver.bidi.bluetooth.request_device_prompt_updated.subscribe):
(window.test_driver.bidi.bluetooth.request_device_prompt_updated):
(window.test_driver.bidi.bluetooth.):
(window.test_driver.bidi.):
(window.test_driver):
(window.test_driver.set_test_context): Deleted.
(window.test_driver.message_test): Deleted.
(window.test_driver.bless): Deleted.
(window.test_driver.click): Deleted.
(window.test_driver.delete_all_cookies): Deleted.
(window.test_driver.get_all_cookies): Deleted.
(window.test_driver.async let): Deleted.
(window.test_driver.get_named_cookie): Deleted.
(window.test_driver.send_keys): Deleted.
(window.test_driver.freeze): Deleted.
(window.test_driver.minimize_window): Deleted.
(window.test_driver.set_window_rect): Deleted.
(window.test_driver.get_window_rect): Deleted.
(window.test_driver.action_sequence): Deleted.
(window.test_driver.generate_test_report): Deleted.
(window.test_driver.set_permission): Deleted.
(window.test_driver.add_virtual_authenticator): Deleted.
(window.test_driver.remove_virtual_authenticator): Deleted.
(window.test_driver.add_credential): Deleted.
(window.test_driver.get_credentials): Deleted.
(window.test_driver.remove_credential): Deleted.
(window.test_driver.remove_all_credentials): Deleted.
(window.test_driver.set_user_verified): Deleted.
(window.test_driver.set_storage_access): Deleted.
(window.test_driver.set_spc_transaction_mode): Deleted.
(window.test_driver.set_rph_registration_mode): Deleted.
(window.test_driver.cancel_fedcm_dialog): Deleted.
(window.test_driver.click_fedcm_dialog_button): Deleted.
(window.test_driver.select_fedcm_account): Deleted.
(window.test_driver.get_fedcm_account_list): Deleted.
(window.test_driver.get_fedcm_dialog_title): Deleted.
(window.test_driver.get_fedcm_dialog_type): Deleted.
(window.test_driver.set_fedcm_delay_enabled): Deleted.
(window.test_driver.reset_fedcm_cooldown): Deleted.
(window.test_driver.create_virtual_sensor): Deleted.
(window.test_driver.update_virtual_sensor): Deleted.
(window.test_driver.remove_virtual_sensor): Deleted.
(window.test_driver.get_virtual_sensor_information): Deleted.
(window.test_driver.set_device_posture): Deleted.
(window.test_driver.clear_device_posture): Deleted.
(window.test_driver_internal.async click): Deleted.
(window.test_driver_internal.async delete_all_cookies): Deleted.
(window.test_driver_internal.async get_all_cookies): Deleted.
(window.test_driver_internal.async get_named_cookie): Deleted.
(window.test_driver_internal.): Deleted.
(window.test_driver_internal.async send_keys): Deleted.
(window.test_driver_internal.async freeze): Deleted.
(window.test_driver_internal.async minimize_window): Deleted.
(window.test_driver_internal.async set_window_rect): Deleted.
(window.test_driver_internal.async get_window_rect): Deleted.
(window.test_driver_internal.async action_sequence): Deleted.
(window.test_driver_internal.async generate_test_report): Deleted.
(window.test_driver_internal.async set_permission): Deleted.
(window.test_driver_internal.async add_virtual_authenticator): Deleted.
(window.test_driver_internal.async remove_virtual_authenticator): Deleted.
(window.test_driver_internal.async add_credential): Deleted.
(window.test_driver_internal.async get_credentials): Deleted.
(window.test_driver_internal.async remove_credential): Deleted.
(window.test_driver_internal.async remove_all_credentials): Deleted.
(window.test_driver_internal.async set_user_verified): Deleted.
(window.test_driver_internal.async set_storage_access): Deleted.
(window.test_driver_internal.async set_spc_transaction_mode): Deleted.
(window.test_driver_internal.set_rph_registration_mode): Deleted.
(window.test_driver_internal.async cancel_fedcm_dialog): Deleted.
(window.test_driver_internal.async click_fedcm_dialog_button): Deleted.
(window.test_driver_internal.async select_fedcm_account): Deleted.
(window.test_driver_internal.async get_fedcm_account_list): Deleted.
(window.test_driver_internal.async get_fedcm_dialog_title): Deleted.
(window.test_driver_internal.async get_fedcm_dialog_type): Deleted.
(window.test_driver_internal.async set_fedcm_delay_enabled): Deleted.
(window.test_driver_internal.async reset_fedcm_cooldown): Deleted.
(window.test_driver_internal.async create_virtual_sensor): Deleted.
(window.test_driver_internal.async update_virtual_sensor): Deleted.
(window.test_driver_internal.async remove_virtual_sensor): Deleted.
(window.test_driver_internal.async get_virtual_sensor_information): Deleted.
(window.test_driver_internal.async set_device_posture): Deleted.
(window.test_driver_internal.async clear_device_posture): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness-shadowrealm-audioworkletprocessor.js: Added.
(globalThis.TestRunner):
(globalThis.TestRunner.prototype.fetchOverPortExecutor):
(globalThis.TestRunner.prototype.async createShadowRealmAndStartTests):
(globalThis.TestRunner.prototype.process):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness-shadowrealm-inner.js: Added.
(globalThis.setShadowRealmGlobalProperties):
(globalThis.GLOBAL.isWindow):
(globalThis.GLOBAL.isWorker):
(globalThis.GLOBAL.isShadowRealm):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness-shadowrealm-outer.js: Added.
(globalThis.shadowRealmEvalAsync):
(async globalThis):
(async if):
(globalThis.getPostMessageFunc):
(else):
(realm.evaluate):
(globalThis.setupFakeDynamicImportInShadowRealm):
(globalThis.setupFakeFetchOverMessagePort):
(globalThis.createSetupErrorResult):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness.js:
(WindowTestEnvironment):
(WindowTestEnvironment.prototype.setup_messages):
(WindowTestEnvironment.prototype.test_timeout):
(ShellTestEnvironment):
(ShellTestEnvironment.prototype.next_default_test_name):
(is_service_worker):
(promise_rejects_js):
(promise_rejects_dom):
(promise_rejects_exactly):
(this.wait_for):
(this.stop_watching):
(EventWatcher):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharnessreport.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/web-bluetooth-bidi-test.js: Added.
(string_appeared_here.convertToBidiManufacturerData):
(FakeBluetooth):
(FakeBluetooth.prototype.async simulateCentral):
(FakeCentral):
(FakeCentral.prototype.async simulatePreconnectedPeripheral):
(FakeCentral.prototype.fetchOrCreatePeripheral_):
(FakePeripheral):
(initializeBluetoothBidiResources):
* LayoutTests/resources/testharness.js:
(WindowTestEnvironment):
(WindowTestEnvironment.prototype._dispatch):
(WindowTestEnvironment.prototype.setup_messages):
(WindowTestEnvironment.prototype.test_timeout):
(ShadowRealmTestEnvironment):
(ShadowRealmTestEnvironment.prototype.begin):
(ShadowRealmTestEnvironment.prototype.add_on_loaded_callback):
(ShellTestEnvironment):
(ShellTestEnvironment.prototype.next_default_test_name):
(create_test_environment):
(is_service_worker):
(promise_rejects_js):
(promise_rejects_dom):
(promise_rejects_exactly):
(this.wait_for):
(this.stop_watching):
(EventWatcher):
(promise_setup):
(done): Deleted.
(generate_tests): Deleted.
(step_timeout): Deleted.
(truncate): Deleted.
(is_node): Deleted.
(format_value): Deleted.

Replace the now outdated copy under LayoutTests/resources.

* LayoutTests/tests-options.json:
